src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -176,7 +172,7 @@ public SolutionState WithNewWorkspace(Workspace workspace, int workspaceVersion)
         /// <summary>
         /// The Workspace this solution is associated with.
         /// </summary>
-        public Workspace Workspace => _solutionServices.Workspace;
+        public Workspace Workspace => Services.Workspace;
```

Comment:
>public Workspace Workspace => Services.Workspace; [](http://example.com/codeflow?start=8&length=49)

For follow-up: Remove to avoid leaking Workspace here.

Reply:
yup.